addons/pos_coupon/static/src/js/ProductScreen.js

```diff
@@ -16,6 +16,27 @@ odoo.define('pos_coupon.ProductScreen', function (require) {
             _onCouponScan(code) {
                 this.currentOrder.activateCode(code.base_code);
             }
+            async _updateSelectedOrderline(event) {
+                const selectedLine = this.currentOrder.get_selected_orderline();
+                if (selectedLine && selectedLine.is_program_reward && event.detail.key === 'Backspace') {
```

Comment:
only with backspace, or also when we try to set it to 0

Reply:
I believe this is okay since `_setValue` is overridden to not do anything on the reward line unless the user is trying to delete it.